addons/hr_recruitment_integration_website/models/hr_job.py

```diff
@@ -0,0 +1,125 @@
+    r'</div>',
+    re.DOTALL
+).findall
+findall_close_i_tag = re.compile(r'</i>').findall
+DEFAULT_OLG_ENDPOINT = 'https://olg.api.odoo.com'
+AI_BASE_PROMPT = '''IGNORE PREVIOUS PROMPTS AND FORMAT THE JOB AS HTML AND NO HYPERLINKS
+You are a recruiter for a company that is looking to hire a new employee.
+You have been tasked with creating a job description for a new job opening.
+Given the following infos, make a professional, friendly and attractive job description that could be used
+to attract potential candidates (The platforms could be things like LinkedIn, Indeed, Glassdoor, Monster, etc.):
+Here is the data to generate the job description:
+'''
+
+
+class Job(models.Model):
+    _inherit = 'hr.job'
+
+    def _extract_job_webpage_data(self):
+        def extract_section_data(section, section_class):
+            if 'image' in section_class:
+                return ''
+
+            extracted_text = ''
+            if 'comparisons' in section_class:
+                boxes = findall_box(section)
+                for title, content in boxes:
+                    extracted_text += '\n' + title + ':\n'
+                    content = content.split('\n')
+                    for item in content:
+                        extracted_text += sub_tags('', item) + '\n'
+
+            elif 's_rating' in section:
+                ratings = findall_rating(section)
+                extracted_text += 'job features: \n'
+                for rating in ratings:
+                    active_icons = len(findall_close_i_tag(rating[1]))
+                    inactive_icons = len(findall_close_i_tag(rating[2]))
+                    infos = rating[0] + ' ' + str(active_icons) + '/' \
+                        + str(inactive_icons + active_icons) + '\n'
+                    extracted_text += infos
+
+                paragraph_text = findall_paragraph(section)
+                extracted_text += '\njob description: \n'
+                for paragraph in paragraph_text:
+                    paragraph = re.sub(r'\n +', ' ', paragraph)
+                    extracted_text += sub_tags('', paragraph) + '\n'
+            else:
+                extracted_text += sub_tags('', section)
+
+            return re.sub(r'\n+', r'\n', re.sub(' +', ' ', extracted_text)) + '\n\n'
+
+        self.ensure_one()
+        related_webpage = self.website_description
+        url = self.full_url
+        sections = findall_section(related_webpage)
+        post_text = 'job title: ' + self.name + '\njob url: ' + url + '\n\n' \
+                    + self.env.company.name + '\n\n'
+        for section_class, section in sections:
+            post_text += extract_section_data(section, section_class)
+        return post_text
+
+    def _generate_post(self):
+        self.ensure_one()
+        if not self.is_published:
+            raise UserError(_('You can only generate a post for a published job offer.'))
+        prompt = AI_BASE_PROMPT + self._extract_job_webpage_data()
```

Comment:
Not sure if this is overkill or not :thinking: What source are we using to extract the data?

Reply:
so basically I get the website content that is already in self and then I parse the webpage into its section and I try to match every section with the regex to extract only the relevent data from it

Reply:
but yeah it's not pretty to look at

Reply:
but do we want to do that at all? It sounds like a terrible inefficient way to make soul-less job descriptions that will contains hallucinations.
I mean they already wrote it to be published on hr.job, why does it needs to go through those indirections?

Reply:
Actually yes we want that because the thing is that what they wrote is a full webpage and most job boards will not accept that. And actually, I would argue that the generated post is way better than most of the job post that I could see on monster or linkedin for two reasons
1 if they are "good" they are basically the default template of the job board + the intro of the values of the company
2 a non negligeable part of them is just a copy paste of the website content

The best exemple of that is literrally our company 
the post
https://www.monster.com/jobs/search?q=odoo&where=Remote&page=1&so=m.h.s
is literally a ctrl a + c + v of
https://www.odoo.com/fr_FR/jobs/technical-content-writer-sf-ca-buffalo-ny-527

And actually the funiest part is that we didn't even bother to remove artifacts
![Screenshot_20240906_153259](https://github.com/user-attachments/assets/54558f9d-9ba9-4c49-a144-ceeca8c50e19)

So to be honest as an applicant I would rather have an AI post like this one (little bit reworked by the HR) than this

Same job but with chatgpt:
```md
**Job Title:** Technical Content Writer  
**Location:** San Francisco, CA / Buffalo, NY (Hybrid)  
**Department:** Marketing  

**About Odoo:**

Odoo is a leading open-source ERP software company with a global reach. Our suite of integrated business applications supports over 12 million users worldwide in streamlining their operations. With more than 60% year-over-year growth and a network spanning 120+ countries, Odoo’s innovative approach to enterprise resource planning makes it one of the fastest-growing software companies in the world.

We are a profitable and expanding company seeking to hire a **Technical Content Writer** to join our dynamic Marketing Team. If you’re passionate about technology and content creation, and you have the ability to break down complex technical information for diverse audiences, this is your opportunity to thrive in a fast-paced, startup-like environment.

**Responsibilities:**

- Collaborate with cross-functional teams (R&D, Product, Technical, Marketing) to create high-quality technical content for global audiences.
- Develop long-form technical guides, eLearning scripts, product tours, whitepapers, and video tutorials to educate users on Odoo’s suite of business applications.
- Author and manage technical documentation using RST markup and Git/GitHub workflows.
- Adapt content created by Odoo’s marketing team for U.S. audiences, ensuring consistency and precision.
- Lead projects independently using Odoo’s Project management tool, adhering to content calendars and brand guidelines.
- Contribute to Odoo’s mission by enhancing user education through clear, concise, and accurate content.

**What We’re Looking For:**

- **Technical Expertise:** Strong technical writing background with the ability to communicate complex information clearly and simply.
- **Autonomous Learner:** Quick to grasp new technologies and processes; passionate about software and ERP systems.
- **Team Player:** You thrive in a collaborative environment and foster a culture of open communication and knowledge-sharing.
- **Adaptable:** Comfortable taking on new challenges and working in a fast-paced, evolving environment.
  
**Must-Have:**

- Bachelor’s degree in Writing, Communications, Computer Science, or related field (or equivalent experience).
- Proven portfolio of technical writing samples.
- Experience with ERP systems or working in a SaaS company.
- Familiarity with Git/GitHub or the eagerness to learn.
- Hands-on experience writing technical documentation.

**Nice to Have:**

- Multilingual proficiency.
- Experience with A/V content production.
- Immediate availability.

**What’s Great About Working at Odoo:**

- **Culture:** Friendly, open, and supportive work environment.
- **Autonomy:** Take ownership of your projects without rigid processes or excessive bureaucracy.
- **Growth:** Expand your expertise in various industries while contributing to meaningful projects.
- **Benefits:** Healthcare, Dental, Vision, FSA/HSA Matching, 401K Matching, Paid Time Off (PTO), and Commuter Benefits.
- **Perks:** Chef-prepared lunches, snacks, coffee, company-sponsored events, telehealth mental health services, and a $100 contribution to your home office setup.
- **Compensation:** Competitive salary. Estimated annual compensation ranges from $70,000 - $85,000 in San Francisco and $55,000 - $70,000 in Buffalo, with variation based on experience and location.

**Diversity and Inclusion:**

Odoo is committed to creating a diverse and inclusive workplace where everyone feels valued. We encourage people of all backgrounds, experiences, and perspectives to apply.

**Ready to Join Our Team?**

Apply today to become part of a fast-growing, innovative company where your contributions truly make an impact! We look forward to learning more about you.

---

Note: This is a hybrid role that requires working 3 days in the office and 2 days remotely in either San Francisco, CA or Buffalo, NY. Odoo is unable to sponsor work visas at this time.

**Location:**  
Odoo Inc.  
8000 Marina Blvd #300, Brisbane, CA 94005, United States
```

Reply:
Note I simplify as much as possible the process to make it more maintainable and add clear hints to the user that the data will be send to the AI